### DISCUSSION TITLE
How to contribute

### DISCUSSION THREAD

**Discussion by detrin** [shap/shap#3049] [Q&A] [answered]
As I use this package on an almost daily basis at work but also in my personal project. I would like to contribute some of my time to this project. Would you suggest what would be a good way of doing so? I have years of history coding in Python, so I feel confident in this area. 

Since I see the new initiative during the past weeks, I decided to ask new maintainers.

**Comment by connortann** [comment, marked as answer]
Hi @detrin , thanks for getting in touch! We'd definitely love to have more contributors on board.

I think a good place to start would be helping fix up the notebooks, as described in #3036 . You could look at #3037 as an example PR that implements the fixes we're looking for.

I'm keen to make it easier for new contributors to get involved, as we'd like to grow the pool of contributors and maintainers. I recently added a [contributing guide](https://github.com/slundberg/shap/blob/master/CONTRIBUTING.md), which should help get started.

On another note, we have a lot of work to do to clean up the issue tracker (as per discussion in #3051 ) : we could perhaps create a label `good first issue` to highlight areas where new contributors could help out.

**Reply by detrin** [reply]
Thanks @connortann, I will have a look at the notebooks first then. 

I would also suggest to create discord server as a communication channel. I think this will lower the boundary of incoming suggestions and contributors. It can be moderated and contributors would get easily notified. I think it has potential to save some time globally on this project.

**Reply by detrin** [reply]
@thatlittleboy @dsgibbons @connortann @owenlamont @Z4K4RI4 @arshiaar 

I believe that creating discord server or slack would help to build a community around shap. Would you support this idea, you don't mind/don't care or you don't think that it would help much? Many projects have discord server, e.g. https://llvm.org

**Reply by thatlittleboy** [reply]
@detrin In my mind the benefit of having discord/slack is for near-real-time (synchronous) support and discussions.
Conducting our discussions asynchronously on Github and/or StackOverflow means our discussions are persistent and searchable for others facing the same issues.
I'm not sure we have such a strong need for the former kind of communication (yet), to be honest.

One other concern I have is that it is just one more community to moderate and attend to. For me, I'm mostly constrained by my free time to manage this project, not so much about not receiving notifications; hence why I'm more for the asynchronicity of the discussions.

That said, if there is sufficient interest amongst the users and the maintainers to have one, I'm not against the idea either way. I just won't be a heavy participant on that forum.

**Reply by dsgibbons** [reply]
I'm with @thatlittleboy on this. I'd prefer to use GitHub exclusively. I don't think this project needs real-time chat, and I like that all of our decision making is publicly documented in one place.

**Reply by znacer** [reply]
I think it is not a good idea to spread discussions over several forums. Solving users issues on slack won't be helpfull for other people and will just end in users asking for the same thing over and over. 
And there is the responsability to moderate what is said on this forum... I don't think one would be happy to receive notification in the middle of the night because some troll from posted toxic content.

**Comment by mohityadav8** [comment]
hello can i know how will u evaluate a contributer
